### PR TITLE
New action Blacklist in Microsoft Defender ATP plugin

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
New action Blacklist in Microsoft Defender ATP plugin

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nDid not find indicator to delete Indicator not deleted",
    "log": "Connecting...\nTime Now: 1596644678.764897\nTime Ago: 0\nSeconds elapsed:1596644678\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************Zhs0Q\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644679.227938\nTime Ago: 1596644679.2253325\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\nAn error occurred during plugin execution!\n\nDid not find indicator to delete Indicator not deleted\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 307, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 428, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/microsoft_atp_rapid7_plugin-4.2.0-py3.8.egg/komand_microsoft_atp/actions/blacklist/action.py\", line 23, in run\n    self._create_or_update_indicator(params) if indicator_state else self._delete_indicator(params)\n  File \"/usr/local/lib/python3.8/site-packages/microsoft_atp_rapid7_plugin-4.2.0-py3.8.egg/komand_microsoft_atp/actions/blacklist/action.py\", line 33, in _delete_indicator\n    raise PluginException(\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nDid not find indicator to delete Indicator not deleted\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/bad_blacklist_delete_not_existing.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644683.6421082\nTime Ago: 0\nSeconds elapsed:1596644683\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************x4xLQ\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644684.0991073\nTime Ago: 1596644684.0956638\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "AlertAndBlock",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-08-05T16:07:50.3782695Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "21",
        "indicatorType": "Url",
        "indicatorValue": "http://google.com",
        "lastUpdateTime": "2020-08-05T16:24:44.5708523Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_default_action.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644687.9848416\nTime Ago: 0\nSeconds elapsed:1596644687\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************b3Pww\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644688.4199538\nTime Ago: 1596644688.4168978\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-07-30T18:04:40.5031044Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "5",
        "indicatorType": "DomainName",
        "indicatorValue": "google.com",
        "lastUpdateTime": "2020-08-05T16:24:48.869632Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "High",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_default_severity.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644692.3610375\nTime Ago: 0\nSeconds elapsed:1596644692\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************6VQig\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644692.8118706\nTime Ago: 1596644692.8088505\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-08-05T16:08:00.2395224Z",
        "description": "http://google.com",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "22",
        "indicatorType": "Url",
        "indicatorValue": "http://google.com",
        "lastUpdateTime": "2020-08-05T16:24:53.1671011Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "http://google.com"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_default_title_and_description.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644696.6596305\nTime Ago: 0\nSeconds elapsed:1596644696\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************GvDvg\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644697.114243\nTime Ago: 1596644697.1112318\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-07-30T18:04:40.5031044Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "5",
        "indicatorType": "DomainName",
        "indicatorValue": "google.com",
        "lastUpdateTime": "2020-08-05T16:24:57.3440732Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_domain.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644700.7880547\nTime Ago: 0\nSeconds elapsed:1596644700\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************tZ_HA\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644701.223252\nTime Ago: 1596644701.2202454\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-07-30T19:02:02.3176007Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "12",
        "indicatorType": "IpAddress",
        "indicatorValue": "82.242.248.2",
        "lastUpdateTime": "2020-08-05T16:25:01.4669221Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_ip.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644705.059696\nTime Ago: 0\nSeconds elapsed:1596644705\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************H79Fg\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644705.796339\nTime Ago: 1596644705.7929094\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-07-30T18:06:27.9384355Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "7",
        "indicatorType": "FileSha1",
        "indicatorValue": "3da541559918a808c2402bba5012f6c60b27661c",
        "lastUpdateTime": "2020-08-05T16:25:06.1858371Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_sha1.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644709.6108088\nTime Ago: 0\nSeconds elapsed:1596644709\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************wgrsA\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644710.0772038\nTime Ago: 1596644710.0741591\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-07-30T18:06:52.8434807Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "8",
        "indicatorType": "FileSha256",
        "indicatorValue": "9a6ec7b822c78850f45a5d87b916469e5c1a24fb8571b8941f5e69e61793f64a",
        "lastUpdateTime": "2020-08-05T16:25:10.5236696Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_sha256.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644714.163596\nTime Ago: 0\nSeconds elapsed:1596644714\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************FlAiw\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644714.605776\nTime Ago: 1596644714.6027138\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "@odata.context": "https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity",
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-08-05T16:08:00.2395224Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "22",
        "indicatorType": "Url",
        "indicatorValue": "http://google.com",
        "lastUpdateTime": "2020-08-05T16:25:15.0627562Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_add_url.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connecting...\nTime Now: 1596644718.4995568\nTime Ago: 0\nSeconds elapsed:1596644718\nRefreshing auth token\nUpdating Auth Token...\nGetting token from: https://login.windows.net/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication was successful, token is: ******************9xyXw\nrapid7/Microsoft Windows Defender ATP:4.2.0. Step name: blacklist\nRunning...\nTime Now: 1596644718.9396853\nTime Ago: 1596644718.9370017\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\nTime Now: 1596644719.7176766\nTime Ago: 1596644718.9370017\nSeconds elapsed:0\nToken is valid, not refreshing.\nUpdating session headers.\n",
    "meta": {},
    "output": {
      "indicator_action_response": {
        "action": "Alert",
        "application": "application",
        "category": 1,
        "createdBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "createdByDisplayName": "WindowsDefenderATPSiemConnector",
        "createdBySource": "PublicApi",
        "creationTimeDateTimeUtc": "2020-08-05T16:08:00.2395224Z",
        "description": "some description",
        "expirationTime": "2020-12-12T00:00:00Z",
        "generateAlert": true,
        "historicalDetection": false,
        "id": "22",
        "indicatorType": "Url",
        "indicatorValue": "http://google.com",
        "lastUpdateTime": "2020-08-05T16:25:15.0627562Z",
        "lastUpdatedBy": "82f42fca-e931-4f03-b54c-47af94bd394d",
        "mitreTechniques": [],
        "rbacGroupIds": [],
        "rbacGroupNames": [],
        "recommendedActions": "nothing",
        "severity": "Informational",
        "source": "WindowsDefenderATPSiemConnector",
        "sourceType": "AadApp",
        "title": "Title"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp:4.2.0 --debug run < tests/blacklist_delete.json
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: help.md[96]: https://api.securitycenter.windows.com/api/$metadata#Indicators/$entity
[*] Plugin successfully validated!

----
[*] Total time elapsed: 13357.093ms
```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: microsoft_atp/plugin.spec.yaml [hunk 1]
```diff
@@ -594,6 +722,100 @@ actions:
...
+    title: Blacklist
+    description: Submit or update new indicator
+    input:
+      indicator:
+        title: Indicator
+        description: The value of the indicator
+        type: string
+        example: 220e7d15b011d7fac48f2bd61114db1022197f7f
+        required: true
+      indicator_type:
```
Comment (jschipp-r7) [inline microsoft_atp/plugin.spec.yaml:735]: We should auto-detect the indicator in the code so the user doesn't have to choose the type.

File: microsoft_atp/plugin.spec.yaml [hunk 2]
```diff
@@ -594,6 +722,100 @@ actions:
...
+        title: Recommended Actions
+        description: TI indicator alert recommended actions
+        type: string
+        example: nothing
+        required: false
+      rbac_group_names:
+        title: RBAC Group Names
+        description: List of RBAC group names the indicator would be applied to
+        type: '[]string'
+        example: [group1,group2]
```
Comment (jschipp-r7) [inline microsoft_atp/plugin.spec.yaml:804]: Needs to be a valid JSON array
```suggestion
        example: '["group1","group2"]'
```

File: microsoft_atp/plugin.spec.yaml (hunk 1 above)
Comment (r7-mgorny) [reply microsoft_atp/plugin.spec.yaml:735]: @jschipp-r7 
If user leave empty this field, type will be autodetected so user doesn't have to choose it. We can remove this field of course if needed :)

Comment (jschipp-r7) [reply microsoft_atp/plugin.spec.yaml:735]: @r7-mgorny yeah, if it's auto-detected we don't need the extra input. Let's remove it - thanks

**Pull request merged**